diff --git a/src/constants.vala.in b/src/constants.vala.in
index 31c976d..dd2b63a 100644
--- a/src/constants.vala.in
+++ b/src/constants.vala.in
@@ -158,7 +158,7 @@ namespace BookwormApp.Constants {
     public const string INSTALL_TASKS_DIR = "@MESON_INSTALL_TASKSDIR@";
     public const string INSTALL_MOBILIB_DIR = "@MESON_INSTALL_MOBILIBDIR@";
     public const string RESOURCE_PREFIX = "/com/github/babluboy/bookworm/";
-    public const string EBOOK_EXTRACTION_LOCATION = "/tmp/" + bookworm_id + "/";
+    public const string EBOOK_EXTRACTION_LOCATION = "@TERMUX_PREFIX@/tmp/" + bookworm_id + "/";
     public const string PREFIX_FOR_FILE_URL = "file:///";

